### PR TITLE
[SQL Server] (No.3) add firewall rule related codes

### PR DESCRIPTION
**Pull request by shenqianjin**

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java [hunk 1]
```diff
@@ -61,17 +62,15 @@ public void delete() {
...
     }
 
-    private com.azure.resourcemanager.sql.models.SqlServer getSqlServerInner() {
-        if (sqlServerInner == null) {
-            refreshWebAppInner();
-        }
-        return sqlServerInner;
+    @Override
+    public List<SqlFirewallRuleEntity> firewallRules() {
+        return sqlServerInner.firewallRules().list().stream().map(e -> formSqlServerFirewallRule(e)).collect(Collectors.toList());
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java:73]: use lambda e -> formSqlServerFirewallRule(e)

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java [hunk 2]
```diff
@@ -113,56 +121,41 @@ public SqlServer commit() {
...
-            // update entity properties after created sql server successfully.
-            SqlServer.this.entity = SqlServer.this.fromSqlServer(server);
             // update
             if (isEnableAccessFromAzureServices() || isEnableAccessFromLocalMachine()) {
-                new SqlServerFirewallUpdater().withEnableAccessFromAzureServices(isEnableAccessFromAzureServices())
-                    .withEnableAccessFromLocalMachine(isEnableAccessFromLocalMachine()).commit();
+                SqlServer.this.update().commit();
             }
+            // refresh entity
+            SqlServer.this.refreshWebAppInner();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java:129]: why we need to refresh webapp inner?

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java [hunk 3]
```diff
@@ -113,56 +121,41 @@ public SqlServer commit() {
...
-            } else if (serverEnableAccessFromAzureServices(firewallRuleList)) {
+            } else {
                 sqlServerInner.removeAccessFromAzureServices();
             }
+            // update common rule
             if (isEnableAccessFromLocalMachine()) {
-                enableAccessFromLocalMachine();
-            } else if (serverEnableAccessFromLocalMachine(firewallRuleList)) {
-                removeAccessFromLocalMachine();
+                SqlFirewallRuleEntity firewallEntity = SqlFirewallRuleEntity.builder().name("").startIpAddress("").endIpAddress("").build();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java:146]: this is a partials code?

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java (hunk 2 above)
Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java:135]: update Sql server entity and inner object once any property changed.

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java (hunk 3 above)
Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java]: Yes. updated now.

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java (hunk 1 above)
Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java]: updated.

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java [hunk 4]
```diff
@@ -143,15 +151,57 @@ public SqlServer commit() {
...
+                            + "trustServerCertificate=false;"
+                            + "loginTimeout=30;";
+            connectionUrl = String.format(connectionUrl, entity.getName(), entity.getAdministratorLoginName(), entity.getName());
+            ;
+            try {
+                Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
+                DriverManager.getConnection(connectionUrl);
+                // Code here.
+            } catch (SQLException e) {
+                e.printStackTrace();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java:194]: remove this:e.printStackTrace()

Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java]: removed.

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/CheckNameAvailabilityResultEntity.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+package com.microsoft.azure.toolkit.lib.sqlserver.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public class CheckNameAvailabilityResultEntity {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/CheckNameAvailabilityResultEntity.java:13]: I think this class is common and should be shared(not in the sqlserver.model package)

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/NetUtils.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.regex.Pattern;
+
+public class NetUtils {
+
+    private static final Pattern IPADDRESS_PATTERN = Pattern.compile("\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}");
+
+    public static String getPublicIp() {
+        String ip = StringUtils.EMPTY;
+        try {
+            final URL url = new URL("https://ipecho.net/plain");
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/NetUtils.java:30]: duplicate code as AzureMySQLService

Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/NetUtils.java:30]: MySQL should use these common's one finally.

**Review by andxu — APPROVED** [review]

**Pull request merged**